### PR TITLE
feat(menu): increase nested menu elevation based on depth

### PR DESCRIPTION
**Pull request by crisbeto**
Increases the sub-menu elevation, based on its depth. [Spec for reference](https://material.io/guidelines/material-design/elevation-shadows.html)

### CODE REVIEW

File: src/lib/menu/menu-directive.ts [hunk 1]
```diff
@@ -62,8 +63,8 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
   /** Current state of the panel animation. */
   _panelAnimationState: 'void' | 'enter-start' | 'enter' = 'void';
 
-  /** Whether the menu is a sub-menu or a top-level menu. */
-  isSubmenu: boolean = false;
+  /** Parent menu os the current menu panel. */
```
Comment (kara) [inline src/lib/menu/menu-directive.ts]: Typo: `os` -> `of`

File: src/lib/menu/menu-directive.ts [hunk 2]
```diff
@@ -181,6 +182,25 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
     this._classList['mat-menu-below'] = posY === 'below';
   }
 
+  /**
+   * Sets the menu panel elevation.
+   * @param depth Amount of parent menus that come before the menu.
```
Comment (kara) [inline src/lib/menu/menu-directive.ts]: Nit: `Amount of parent menus` -> `Number of parent menus` (sounds more natural)

File: src/lib/menu/menu-directive.ts [hunk 3]
```diff
@@ -181,6 +182,25 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
...
+   */
+  setElevation(depth: number): void {
+    // The elevation starts at 2 and increases by one for each level.
+    const newElevation = `mat-elevation-z${depth + 2}`;
```
Comment (kara) [inline src/lib/menu/menu-directive.ts]: Can we make the starting elevation a constant?  Like MD_MENU_BASE_ELEVATION

**Comment by andrewseguin** [comment]
Please rebase

**Comment by crisbeto** [comment]
Rebased @andrewseguin.

**Pull request merged**